From 5518dd64224c3dfe2674a496c771770904458a78 Mon Sep 17 00:00:00 2001
From: electimon <electimon@gmail.com>
Date: Tue, 01 Apr 2025 14:00:10 -0400
Subject: [PATCH] HintManagerService: Generate dummy mSupportInfo for non power AIDL devices

* Before it would generate dummy info when the IPower AIDL version
  was too old. Now we will generate it when IPower is just completely
  unavailable as well, avoiding a crash from accessing an
  uninitialized mSupportInfo.

test: sailfish boot :D
Change-Id: Ibf43d0bfea9fd6cdeae35248e4dde58c01d3ef3b
Signed-off-by: electimon <electimon@gmail.com>
---

diff --git a/services/core/java/com/android/server/power/hint/HintManagerService.java b/services/core/java/com/android/server/power/hint/HintManagerService.java
index 1cf24fc..b84ddc9 100644
--- a/services/core/java/com/android/server/power/hint/HintManagerService.java
+++ b/services/core/java/com/android/server/power/hint/HintManagerService.java
@@ -335,6 +335,8 @@
             } catch (RemoteException e) {
                 throw new IllegalStateException("Could not contact PowerHAL!", e);
             }
+        } else {
+            mSupportInfo = getDummySupportInfo();
         }
         mDefaultCpuHeadroomCalculationWindowMillis =
                 new CpuHeadroomParamsInternal().calculationWindowMillis;
@@ -379,7 +381,10 @@
         } catch (RemoteException e) {
             throw new IllegalStateException("Could not contact PowerHAL!", e);
         }
+        return getDummySupportInfo();
+    }
 
+    private SupportInfo getDummySupportInfo() {
         SupportInfo supportInfo = new SupportInfo();
         supportInfo.usesSessions = isHintSessionSupported();
         // Global boosts & modes aren't currently relevant for HMS clients
